crypto/bio/bss_conn.c

```diff
@@ -544,12 +540,8 @@ static long conn_ctrl(BIO *b, int cmd, long num, void *ptr)
         break;
 # ifndef OPENSSL_NO_KTLS
     case BIO_CTRL_SET_KTLS:
-#  ifdef __FreeBSD__
-        crypto_info = (struct tls_enable *)ptr;
-#  else
-        crypto_info = (struct tls12_crypto_info_aes_gcm_128 *)ptr;
-#  endif
-        ret = ktls_start(b->num, crypto_info, sizeof(*crypto_info), num);
+        crypto_info = (ktls_crypto_info_t *)ptr;
+        ret = ktls_start(b->num, crypto_info, num);
```

Comment:
If you don't mind the implicit pointer cast, we could just pass `ptr` directly to `ktls_start` and remove `crypto_info` entirely both here and in bss_sock.c.

Reply:
While I'm sure we could find precedent for such behavior in the tree if we went looking, I don't feel a need to push to make things so concise, and am fine keeping `crypto_info` around.